cfme/infrastructure/provider/scvmm.py

```diff
@@ -67,10 +58,108 @@ def from_config(cls, prov_config, prov_key, appliance=None):
             sec_protocol=prov_config['sec_protocol'],
             sec_realm=prov_config['sec_realm'],
             appliance=appliance)
+
+    def create(self, cancel=False, validate_credentials=True, check_existing=False,
+               validate_inventory=False):
+        """
+        Creates a provider in the UI
+
+        Args:
+            cancel (boolean): Whether to cancel out of the creation.  The cancel is done
+                after all the information present in the Provider has been filled in the UI.
+            validate_credentials (boolean): Whether to validate credentials - if True and the
+                credentials are invalid, an error will be raised.
+            check_existing (boolean): Check if this provider already exists, skip if it does
+            validate_inventory (boolean): Whether or not to block until the provider stats in CFME
+                match the stats gleaned from the backend management system
+                (default: ``False``)
+
+        Returns:
+            True if it was created, False if it already existed
+        """
+        if check_existing and self.exists:
+            created = False
+        else:
+            created = True
+
+            logger.info('Setting up provider: %s', self.key)
+            view = navigate_to(self, 'Add')
+
+            # filling main part of dialog
+            view.fill(self)
+
+            # filling endpoints
+            endpoint = self.endpoints[0]
+            # endpoints view contains fields only when there is only endpoint
+            view.endpoints.fill(endpoint)
+
+            # filling credentials
+            view.endpoints.fill(endpoint.credentials)
+            if validate_credentials:
+                view.endpoints.validate.click()
+
+            if cancel:
+                view.cancel.click()
+            else:
+                view.add.click()
+
+            if not cancel:
+                # todo: replace flash when main view is merged
+                flash.assert_message_match('{} Providers "{}" was saved'.format(self.string_name,
+                                                                                self.name))
+        if validate_inventory:
+            self.validate()
+
+        return created
+
+    def update(self, updates=None, endpoints=None, cancel=False, validate_credentials=True):
+        """
+        Updates a provider in the UI.  Better to use utils.update.update context
+        manager than call this directly.
+
+        Args:
+           updates (dict): fields that are changing.
+           endpoints (VirtualCenterEndpoint):
+           cancel (boolean): whether to cancel out of the update.
+        """
+        view = navigate_to(self, 'Edit')
+        # todo: to replace/merge this code with create
+        # update values:
+        # filling main part of dialog
+        if updates:
+            default_val = self.as_fill_value()
+            default_val.update(updates)
+            view.fill(default_val)
```

Comment:
Hmm, this looks like it would deserve DRY - why do we need the original values here? WT view fill only touches things that are specified in the dictionary and nothing else.